### PR TITLE
perf(linter): use `CompactStr` in `no-bitwise`

### PR DESCRIPTION
**Pull request by camchenry**

### CODE REVIEW

**Comment by camchenry** [comment]
* **#6402** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6402?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6401** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6401?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6400** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6400?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6399** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6399?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6398** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6398?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @camchenry and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_linter/src/rules/eslint/no_bitwise.rs [hunk 1]
```diff
@@ -57,10 +57,7 @@ impl Rule for NoBitwise {
                 .and_then(|v| v.get("allow"))
                 .and_then(serde_json::Value::as_array)
                 .map(|v| {
-                    v.iter()
-                        .filter_map(serde_json::Value::as_str)
-                        .map(ToString::to_string)
-                        .collect()
+                    v.iter().filter_map(serde_json::Value::as_str).map(CompactStr::from).collect()
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/eslint/no_bitwise.rs:60]: non-blocking: can we also sort this list then use a binary search in `allowed_operator`?

**Review by DonIsaac — APPROVED** [review]

File: crates/oxc_linter/src/rules/eslint/no_bitwise.rs (hunk 1 above)
Comment (camchenry) [reply crates/oxc_linter/src/rules/eslint/no_bitwise.rs:60]: I think there's a lot of these that could be rewritten, I'll see if I can batch them together

Comment (camchenry) [reply crates/oxc_linter/src/rules/eslint/no_bitwise.rs:60]: @DonIsaac I think this is a bigger discussion, I posted some thoughts in https://github.com/oxc-project/oxc/issues/6408

**Comment by camchenry** [comment]
### Merge activity

* **Oct 9, 7:44 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

**Pull request merged**